packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -501,6 +502,13 @@ export interface AddApplicationTargetsProps extends AddRuleProps {
    */
   readonly port?: number;
 
+  /**
+   * The port used to attach to targets.
```

Comment:
Also, it seems like `targetPort` can be hostport or container port, which might lead to confusion. It would be better to only specify container port number (since users might care more about the container port when building microservices).

Reply:
TargetPort must strictly be the ingress (host) port in this design since it must be accessible from outside the container by a non-ECS resource; as you noted above the container port is potentially not unique in a multi-container task definition. The mapping to host port allows resolution of these potential collisions. If the user specified the container port here the underlying system would need to convert it to a host port anyway.